rpc/src/router.rs

```diff
@@ -0,0 +1,153 @@
+use std::{collections::HashMap, marker::PhantomData};
+
+use serde::{Deserialize, Serialize};
+
+pub struct Router {
+    pub routes: HashMap<String, Box<dyn Handler + 'static + Send + Sync>>,
```

Comment:
Consider running the Router in a separate thread and using a single-thread per-core runtime like tokio-uring. This will make the code and types significantly less complex - you will be able to remove 'static, Send and Sync while still using an async runtime.

https://emschwartz.me/async-rust-can-be-a-pleasure-to-work-with-without-send-sync-static/